### ISSUE TITLE
PivotRowTotals settings not respected when rowdata changes

### ISSUE THREAD

**Issue by SietsevTilburg** [ag-grid/ag-grid#8326]
**I'm submitting a**
```
[x] bug report => see 'Providing a Reproducible Scenario'
[] feature request => do not use Github for feature requests, see 'Customers of AG Grid'
[] support request => see 'Requesting Community Support'
```

**Providing a Reproducible Scenario**
Here is a plunker of the issue https://plnkr.co/edit/bv6f7ZQt3SFHV1bx?open=main.js&preview
To reproduce the issue I included 2 buttons to empty and fill the rowdata.
1. Clicking empty rowdata button. This will clear the rowdata
2. Clicking fill rowdata button. This will fill the rowdata again

When clicking these button in order 1 -> 2 you will first see when pressing 1 that the total column is still visible.
After pressing 2 the data will return but the total column is now the first column, but the setting for the total is set to after.

When refreshing the preview the total column is again after all the data, were it should be.

**Current behavior**
Currently the setting pivotRowTotals = "after" is not being respected when rowdata is changed to empty and is filled again.
When empty and filled again the pivotRowTotals is getting shown before as if the setting is pivotRowTotals = "before"

**Expected behavior**
I expect that when the data is filled again that the pivotRowTotals settings should still be respected and show the totals after all the data.

**Please tell us about your environment:**

* **AG Grid version:** : 31.3.1 and 32.0.1 (Tested on these versions)

* **Browser:**
Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/125.0.0.0 Safari/537.36

* **Language:** [all | Javascript | vue2/3]

**Comment by AG-Zoheil** [comment]
Hello 

Thank you for reporting this - much appreciated!

Based on your description we've now reproduced this issue as a bug.

We have added this bug to our backlog and we are tracking it with the following reference: **AG-11512**

We try to fix bugs from one release to the next, so this should be fixed in the next release or the one after if it was raised too close to the next release date.

See whether this item will be in the next release by checking the NEXT RELEASE checkbox on the product pipeline page:
https://www.ag-grid.com/ag-grid-pipeline/

The best way to track this is to sign up for AG Grid new release notifications using [the instructions here](https://ag-grid.zendesk.com/hc/en-us/articles/360002648058-How-do-I-sign-up-for-ag-Grid-release-notifications-). This way you'll know as soon as a new version is out and you can check whether this specific item was implemented on [the changelog page](https://www.ag-grid.com/ag-grid-changelog/).

Thanks again for bringing this up with us.

Kind regards,
Zoheil

**Comment by Tim524787** [comment]
@AG-Zoheil, could this issue please be reopened?

This is marked as fixed in v32.2.0, but the bug is still there.

**Comment by AG-Zoheil** [comment]
Hi @Tim524787,

You need to use the following: https://www.ag-grid.com/javascript-data-grid/grid-options/#reference-columns-enableStrictPivotColumnOrder in your grid.

Kind regards,
Zoheil
